### PR TITLE
Webpa replace

### PR DESCRIPTION
**Pull request by vasuki01**

### CODE REVIEW

File: source/broadband/webpa_replace.c [hunk 1]
```diff
@@ -63,30 +65,32 @@ void replaceTable(char *objectName,TableData * list,unsigned int paramcount,WDMP
     }
     else
     {
-        WalPrint("paramName after mapping : %s\n",paramName);
+        WalInfo("paramName after mapping : %s\n",paramName);
```
Comment (shilpa24balaji) [inline source/broadband/webpa_replace.c:68]: Revert the info log changes

File: source/broadband/webpa_replace.c [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 #include "webpa_table.h"
 
+#define RBUS_ENABLE  "Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Feature.RBUS.Enable"
```
Comment (sadhyama) [inline source/broadband/webpa_replace.c:11]: Add this TR181 in a common .h file - webpa_internal.h

File: source/broadband/webpa_replace.c [hunk 3]
```diff
@@ -475,33 +492,38 @@ static int getWritableParams(char *paramName, char ***writableParams, int *param
...
-        WalPrint("val_size : %d, ret : %d\n",val_size,ret);
+        WalInfo("val_size : %d, ret : %d\n",val_size,ret);
         if(ret == CCSP_SUCCESS && val_size > 0)
         {
             cnt1 = 0;
             for(cnt = 0; cnt < val_size; cnt++)
             {
                 len = strlen(paramName);
-                if(parameterInfo[cnt]->writable == 1)
+                if(((strncmp(isRbus, "true", strlen("true")) == 0) && parameterInfo[cnt]->writable == 2) ||
```
Comment (sadhyama) [inline source/broadband/webpa_replace.c:521]: Add null check for isRbus . 
Can we avoid doing multiple string operations for each parameter in for loop, by adding null check and defining a flag variable at the beginning and use that flag to to decide if it is rbus/dbus mode .

File: source/broadband/webpa_replace.c [hunk 4]
```diff
@@ -475,33 +492,38 @@ static int getWritableParams(char *paramName, char ***writableParams, int *param
...
+        WalInfo("val_size : %d, ret : %d\n",val_size,ret);
         if(ret == CCSP_SUCCESS && val_size > 0)
         {
             cnt1 = 0;
             for(cnt = 0; cnt < val_size; cnt++)
             {
                 len = strlen(paramName);
-                if(parameterInfo[cnt]->writable == 1)
+                if(((strncmp(isRbus, "true", strlen("true")) == 0) && parameterInfo[cnt]->writable == 2) || 
+                   ((strncmp(isRbus, "false", strlen("false")) == 0) && parameterInfo[cnt]->writable == 1))
```
Comment (sadhyama) [inline source/broadband/webpa_replace.c:522]: add a comment line also on this change as return value of writable info is different for dbus/rbus modes

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: source/broadband/webpa_replace.c [hunk 5]
```diff
@@ -491,13 +509,16 @@ static int getWritableParams(char *paramName, char ***writableParams, int *param
         WalInfo("parameterName: %s, CompName : %s, dbusPath : %s\n", paramName, ppComponents[0]->componentName, ppComponents[0]->dbusPath);
         ret = CcspBaseIf_getParameterNames(bus_handle,ppComponents[0]->componentName, ppComponents[0]->dbusPath, paramName,1,&val_size,&parameterInfo);
         WalPrint("val_size : %d, ret : %d\n",val_size,ret);
-        if(ret == CCSP_SUCCESS && val_size > 0)
+        if(ret == CCSP_SUCCESS && val_size > 0 && isRbus != NULL)
         {
+	    rbusMode = (strncmp(isRbus, "true", strlen("true")) == 0) ? 1 : 0;
```
Comment (sadhyama) [inline source/broadband/webpa_replace.c:514]: Can you move this check to line 502 just after GET and log an error when isRbus is NULL

File: source/broadband/webpa_replace.c [hunk 6]
```diff
@@ -475,13 +488,18 @@ static int getWritableParams(char *paramName, char ***writableParams, int *param
...
     char temp[MAX_PARAMETERNAME_LEN] = { 0 };
     //TODO: Coverity Requirement stack space exceeds 10000 bytes use 2048bytes forMAX_PARAMETERNAME_LEN.
     char *paramList[MAX_PARAMETERNAME_LEN] = { 0 };
 #if !defined(RDKB_EMU)
     strncpy(l_Subsystem, "eRT.",sizeof(l_Subsystem));
 #endif
     parameterInfoStruct_t **parameterInfo = NULL;
+    
+    isRbus = getParameterValue(RBUS_ENABLE);
+    WalInfo("Rbus =%s\n", isRbus);
```
Comment (sadhyama) [inline source/broadband/webpa_replace.c:502]: log it when isRbus is not null

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: source/broadband/webpa_replace.c [hunk 7]
```diff
@@ -491,13 +509,23 @@ static int getWritableParams(char *paramName, char ***writableParams, int *param
         WalInfo("parameterName: %s, CompName : %s, dbusPath : %s\n", paramName, ppComponents[0]->componentName, ppComponents[0]->dbusPath);
         ret = CcspBaseIf_getParameterNames(bus_handle,ppComponents[0]->componentName, ppComponents[0]->dbusPath, paramName,1,&val_size,&parameterInfo);
         WalPrint("val_size : %d, ret : %d\n",val_size,ret);
+	if(isRbus != NULL)
```
Comment (sadhyama) [inline source/broadband/webpa_replace.c:512]: Move this to line 502 just after GET

File: source/broadband/webpa_replace.c [hunk 8]
```diff
@@ -491,13 +509,23 @@ static int getWritableParams(char *paramName, char ***writableParams, int *param
...
+	{
+		rbusMode = (strncmp(isRbus, "true", strlen("true")) == 0) ? 1 : 0;
+	}
+	else
+	{
+		WalInfo("isRbus is NULL\n");
```
Comment (sadhyama) [inline source/broadband/webpa_replace.c:518]: Add it as WalError

**Review by sadhyama — CHANGES_REQUESTED** [review]

**Review by sadhyama — APPROVED** [review]

**Pull request merged**